### PR TITLE
layers: Add GpuAssistedBase class

### PR DESCRIPTION
**Pull request by jeremyg-lunarg**
This is a common base class for DebugPrintf and GPUAV.

### CODE REVIEW

File: layers/debug_printf.cpp [hunk 1]
```diff
@@ -693,79 +532,14 @@ bool DebugPrintf::CommandBufferNeedsProcessing(VkCommandBuffer command_buffer) {
 
 void DebugPrintf::ProcessCommandBuffer(VkQueue queue, VkCommandBuffer command_buffer) {
     auto cb_node = Get<debug_printf_state::CommandBuffer>(command_buffer);
```
Comment (ncesario-lunarg) [inline layers/debug_printf.cpp:534]: Does this return a shared pointer to  `debug_printf_state::CommandBuffer` or `CMD_BUFFER_STATE` (I know you added some magic to `Get` recently, but can't recall the details). Asking mainly for whether or not the static cast in `ProcessInstrumentationBuffer` is (should be?) necessary?

File: layers/gpu_validation.cpp [hunk 2]
```diff
@@ -1477,6 +1264,48 @@ void GpuAssisted::AnalyzeAndGenerateMessages(VkCommandBuffer command_buffer, VkQ
     memset(debug_output_buffer, 0, sizeof(uint32_t) * words_to_clear);
 }
 
+// For the given command buffer, map its debug data buffers and read their contents for analysis.
+void GpuAssisted::ProcessInstrumentationBuffer(VkQueue queue, CMD_BUFFER_STATE *cb_base) {
```
Comment (ncesario-lunarg) [inline layers/gpu_validation.cpp:1268]: This looks almost identical to the debug printf version? Is there any way to consolidate them?

**Review by ncesario-lunarg — APPROVED** [review]
LGTM, but really depends on Tony.

File: layers/debug_printf.cpp [hunk 3]
```diff
@@ -693,79 +532,14 @@ bool DebugPrintf::CommandBufferNeedsProcessing(VkCommandBuffer command_buffer) {
...
-
-    DispatchQueueWaitIdle(queue);
-
-    for (uint32_t submit_idx = 0; submit_idx < submitCount; submit_idx++) {
-        const VkSubmitInfo2 *submit = &pSubmits[submit_idx];
-        for (uint32_t i = 0; i < submit->commandBufferInfoCount; i++) {
-            ProcessCommandBuffer(queue, submit->pCommandBufferInfos[i].commandBuffer);
-        }
+    for (auto *secondary_cmd_buffer : cb_node->linkedCommandBuffers) {
+        ProcessInstrumentationBuffer(queue, static_cast<debug_printf_state::CommandBuffer *>(secondary_cmd_buffer));
```
[the remark this answers is not in the document]
Comment (jeremyg-lunarg) [reply layers/debug_printf.cpp:539]: yeah I think I can do another pass on this part and get it to be cleaner.

**Review by Tony-LunarG — APPROVED** [review]
LGTM.  Nice re-org!

**Pull request merged**